### PR TITLE
feat: Аdd k6 performance tests for WS server

### LINKED ISSUE #2801: Add k6 performance tests for WS server
### Problem

Currently we have no way of stress/load testing the web socket server.

### Solution

Add configuration which reuses the described scenarios for the server performance tests and execute them against the web socket server.

### Alternatives

_No response_

### LINKED ISSUE #2815: K6 package needs to be updated to use ethers6
### Problem

ethersjs dependency was pumped to v6 but the code syntaxes are still maintained in v5.

### Solution

Update k6 package to use ethers v6

### Alternatives

_No response_

### PR DESCRIPTION
**Pull request by victor-yanev**
**Description**:

This PR introduces performance tests for the WebSocket (WS) server using k6. The tests are designed to ensure the reliability and performance of the WS server under various load conditions.

**Changes:**  

- Fixed `prepare/prep.js` to use `ethers@v6`
- Implemented utility functions in `scenarios/test-ws/common.js`
- Implemented performance tests in the following files:
  - `scenarios/test-ws/eth_chainId.js`
  - `scenarios/test-ws/eth_subscribe_logs.js`
  - `scenarios/test-ws/eth_subscribe_newHeads.js`
  - `scenarios/test-ws/eth_subscribe_newPendingTransactions.js`

**Related issue(s)**:

Fixes #2801

Fixes #2815

**Notes for reviewer**:

Logs from run against testnet:
```bash
     checks.......................................................................: 100.00% ✓ 382       ✗ 0      
     ✓ { scenario:eth_chainId }...................................................: 100.00% ✓ 82        ✗ 0      
     ✓ { scenario:eth_subscribe_logs }............................................: 100.00% ✓ 103       ✗ 0      
     ✓ { scenario:eth_subscribe_newHeads }........................................: 100.00% ✓ 97        ✗ 0      
     ✓ { scenario:eth_subscribe_newPendingTransactions }..........................: 100.00% ✓ 100       ✗ 0      
     data_received................................................................: 3.4 MB  35 kB/s
     data_sent....................................................................: 660 kB  6.9 kB/s
     http_req_blocked.............................................................: avg=94.44ms  min=74.14ms  med=77.94ms max=147.74ms p(90)=126.99ms p(95)=137.37ms
     http_req_connecting..........................................................: avg=35.21ms  min=33.54ms  med=34.99ms max=37.34ms  p(90)=36.87ms  p(95)=37.1ms  
     http_req_duration............................................................: avg=95.1ms   min=72.33ms  med=94.69ms max=118.71ms p(90)=114.17ms p(95)=116.44ms
       { expected_response:true }.................................................: avg=95.1ms   min=72.33ms  med=94.69ms max=118.71ms p(90)=114.17ms p(95)=116.44ms
     ✓ { scenario:eth_chainId,expected_response:true }............................: avg=0s       min=0s       med=0s      max=0s       p(90)=0s       p(95)=0s      
     ✓ { scenario:eth_subscribe_logs,expected_response:true }.....................: avg=0s       min=0s       med=0s      max=0s       p(90)=0s       p(95)=0s      
     ✓ { scenario:eth_subscribe_newHeads,expected_response:true }.................: avg=0s       min=0s       med=0s      max=0s       p(90)=0s       p(95)=0s      
     ✓ { scenario:eth_subscribe_newPendingTransactions,expected_response:true }...: avg=0s       min=0s       med=0s      max=0s       p(90)=0s       p(95)=0s      
     http_req_failed..............................................................: 0.00%   ✓ 0         ✗ 4      
     http_req_receiving...........................................................: avg=8.13ms   min=78µs     med=562µs   max=31.34ms  p(90)=22.24ms  p(95)=26.79ms 
     http_req_sending.............................................................: avg=238.25µs min=57µs     med=105µs   max=686µs    p(90)=513.2µs  p(95)=599.6µs 
     http_req_tls_handshaking.....................................................: avg=40.5ms   min=38.94ms  med=40.81ms max=41.45ms  p(90)=41.36ms  p(95)=41.4ms  
     http_req_waiting.............................................................: avg=86.73ms  min=40.93ms  med=94.04ms max=117.92ms p(90)=113.28ms p(95)=115.6ms 
     http_reqs....................................................................: 4       0.041765/s
     ✗ { scenario:eth_chainId }...................................................: 0       0/s
     ✗ { scenario:eth_subscribe_logs }............................................: 0       0/s
     ✗ { scenario:eth_subscribe_newHeads }........................................: 0       0/s
     ✗ { scenario:eth_subscribe_newPendingTransactions }..........................: 0       0/s
     iteration_duration...........................................................: avg=24.12s   min=759.53ms med=12.1s   max=1m6s     p(90)=1m4s     p(95)=1m5s    
     iterations...................................................................: 1230    12.842846/s
     scenario_duration............................................................: 94949   min=1686    max=94949
     ✓ { scenario:eth_chainId }...................................................: 94949   min=1689    max=94949
     ✓ { scenario:eth_subscribe_logs }............................................: 94844   min=3688    max=94844
     ✓ { scenario:eth_subscribe_newHeads }........................................: 94778   min=1703    max=94778
     ✓ { scenario:eth_subscribe_newPendingTransactions }..........................: 94878   min=1686    max=94878
     vus..........................................................................: 305     min=305     max=400  
     vus_max......................................................................: 400     min=400     max=400  
     ws_connecting................................................................: avg=21.72s   min=254.93ms med=9.46s   max=1m0s     p(90)=1m0s     p(95)=1m0s    
     ws_msgs_received.............................................................: 382     3.988591/s
     ws_msgs_sent.................................................................: 382     3.988591/s
     ws_session_duration..........................................................: avg=21.84s   min=425.42ms med=9.5s    max=1m2s     p(90)=1m0s     p(95)=1m0s    

running (1m35.8s), 000/400 VUs, 1230 complete and 291 interrupted iterations
eth_chainId                    ✓ [======================================] 100 VUs  1m30s
eth_subscribe_logs             ✓ [======================================] 100 VUs  1m30s
eth_subscribe_newHeads         ✓ [======================================] 100 VUs  1m30s
eth_subscribe_newPendingTra... ✓ [======================================] 100 VUs  1m30s
```

**Checklist**

- [ ] Documented (Code comments, README, etc.)
- [ ] Tested (unit, integration, etc.)

### CODE REVIEW

**Comment by quiet-node** [comment]
Added 'Fixes #2815' to the description so the ticket will be closed when this PR is merged.

**Comment by victor-yanev** [comment]
> Added 'Fixes #2815' to the description so the ticket will be closed when this PR is merged.

@quiet-node Great, thank you!

File: k6/src/prepare/contracts/Greeter.sol [hunk 1]
```diff
@@ -0,0 +1,24 @@
+//SPDX-License-Identifier: Unlicense
```
Comment (quiet-node) [inline k6/src/prepare/contracts/Greeter.sol:1]:
```suggestion
//SPDX-License-Identifier: Apache-2.0
```

**Review by quiet-node — APPROVED** [review]

**Review by jeromy-cannon — DISMISSED** [review]

**Review by ebadiere — APPROVED** [review]
LG

**Pull request merged**